src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -20,9 +21,20 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
                 properties.CorrelationId = message.Headers[NServiceBus.Headers.CorrelationId];
             }
 
+            DelayDeliveryWith delayDeliveryWith;
+            DoNotDeliverBefore doNotDeliverBefore;
             DiscardIfNotReceivedBefore timeToBeReceived;
+            long delay = 0;
 
-            if (TryGet(deliveryConstraints, out timeToBeReceived) && timeToBeReceived.MaxTime < TimeSpan.MaxValue)
+            if (TryGet(deliveryConstraints, out delayDeliveryWith))
+            {
+                delay = Convert.ToInt64(Math.Round(delayDeliveryWith.Delay.TotalSeconds) * 1000);
+            }
+            else if (TryGet(deliveryConstraints, out doNotDeliverBefore))
+            {
+                delay = Convert.ToInt64(Math.Round((doNotDeliverBefore.At - DateTime.UtcNow).TotalSeconds) * 1000);
+            }
+            else if (TryGet(deliveryConstraints, out timeToBeReceived) && timeToBeReceived.MaxTime < TimeSpan.MaxValue)
```

Comment:
At the moment I'm just considering these three options mutually exclusive, with delay taking precedence over TTBR.

The other interesting this is that both DelayDeliveryWith and DoNotDeliverBefore can be set on a message, and you just have to pick one or the other to take precedence over the other. How did you handle it in ASB?